examples/snippets/data_api.py

```diff
@@ -0,0 +1,222 @@
+# Copyright 2023 Planet Labs PBC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+"""Example of creating and downloading multiple orders.
+
+This is an example of submitting two orders, waiting for them to complete, and
+downloading them. The orders each clip a set of images to a specific area of
+interest (AOI), so they cannot be combined into one order.
+
+[Planet Explorer](https://www.planet.com/explorer/) was used to define
+the AOIs and get the image ids.
+"""
+from pathlib import Path
+import planet
+from planet import data_filter
+import json
+from datetime import datetime
+
+
+# search
+async def search(item_types, search_filter, name, sort, limit):
+    '''Code snippet for search.'''
+    async with planet.Session() as sess:
+        client = sess.client('data')
+        async for item in client.search(item_types=item_types,
+                                        search_filter=search_filter,
+                                        name=name,
+                                        sort=sort,
+                                        limit=limit):
+            return item
```

Comment:
@kevinlacaille What's the intent here? Why wrap lines 33-40 up in a new function?

Reply:
I believe the intent behind the snippets is to show how to use each of the clients' functions, individually. But please correct me if I'm wrong, @sarasafavi @JuLeeAtPlanet. I'm happy to write these in which ever way we intend to use them.

Reply:
I'm 100% sure that if we show to users that they should wrap 3 SDK statements up in a new function with the same name, it will confuse them as much as it helps. Whenever possible, let's try to show Python SDK usage with no wrappers and no helpers, just to keep things clear. Also returning from an async for loop doesn't quite work.

Reply:
Concrete suggestion using snippet sections https://facelessuser.github.io/pymdown-extensions/extensions/snippets/#snippet-sections.

```suggestion
# --8<-- [start:search]
# Search example.
async with planet.Session() as sess:
    client = sess.client('data')
    async for item in client.search(item_types=item_types, search_filter=search_filter, name=name, sort=sort, limit=limit):
        print(item)
# --8<-- [end:search]
```

Reply:
Yes, this is probably a lot more useful. Also, thanks for adding the embedded markdown stuff!

Reply:
I'll change the snippets to this format 👍